samples/apps/bookinfo/src/productpage/Dockerfile

```diff
@@ -12,10 +12,19 @@
 #   See the License for the specific language governing permissions and
 #   limitations under the License.
 
-FROM python:2-onbuild
+# Ref: https://hub.docker.com/_/python/
+
+FROM python:2
+
+RUN apt-get install libmysqlclient-dev
 
 RUN mkdir -p /opt/microservices
-COPY . /opt/microservices/
-EXPOSE 9080
 WORKDIR /opt/microservices
-CMD python productpage.py 9080
+COPY requirements.txt ./
+RUN pip install --no-cache-dir -r requirements.txt
+
+COPY . .
+EXPOSE 9080
+ARG service_version
+ENV SERVICE_VERSION ${service_version:-v1}
+CMD python productpage.py 9080 $MYSQLADDR
```

Comment:
MYSQLADDR comes from the spec for the productpage Deployment. It can be an IP address or a fqdn. For service_version "v2" the productpage Deployment spec would have the following environment defined for the productpage container.

        env:
        - name: MYSQLADDR
          value: "192.168.144.189"